go/types/value_encoder.go

```diff
@@ -129,7 +130,12 @@ func (w *valueEncoder) writeValue(v Value) {
 	case BoolKind:
 		w.writeBool(bool(v.(Bool)))
 	case NumberKind:
-		w.writeNumber(v.(Number))
+		n := v.(Number)
+		f := float64(n)
+		if math.IsNaN(f) || math.IsInf(f, 0) {
```

Comment:
What about negative zero? Okay to just silently continue and probably encode as zero?